File: iOS/KinKitchen/KinKitchen/Componets/Cards/KinCard.swift
//
//  KinCard.swift
//  KinKitchen
//
//  Created by Greg Hudler on 8/26/26.
//

import SwiftUI

struct KinCard<Content: View>: View {
    let content: Content

    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    var body: some View {
        content
            .padding(KinSpacing.large)
            .background(KinColors.surface)
            .clipShape(
                RoundedRectangle(cornerRadius: KinRadius.large)
            )
    }
}

File: Android/app/src/main/java/com/pushtomaindev/kinkitchen/components/cards/KinCard.kt
package com.pushtomaindev.kinkitchen.components.cards

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import com.pushtomaindev.kinkitchen.ui.theme.KinColors
import com.pushtomaindev.kinkitchen.ui.theme.KinRadius
import com.pushtomaindev.kinkitchen.ui.theme.KinSpacing

/** Mirrors iOS `KinCard`. */
@Composable
fun KinCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit,
) {
    Column(
        modifier = modifier
            .clip(KinRadius.largeShape)
            .background(KinColors.surface)
            .padding(KinSpacing.large),
        content = content,
    )
}
